test/e2e/proxy_publicviewer_test.go

```diff
@@ -0,0 +1,225 @@
+	hostAwait := awaitilities.Host()
+	memberAwait := awaitilities.Member1()
+
+	// public viewer is enabled in ToolchainConfig
+	hostAwait.UpdateToolchainConfig(t, testconfig.PublicViewerConfig(true))
+
+	// we create a space to share and a user for 'community-user'
+	space, _, _ := testsupportspace.CreateSpace(t, awaitilities, testspace.WithTierName("appstudio"), testspace.WithSpecTargetCluster(memberAwait.ClusterName))
+	communityUser := &proxyUser{
+		expectedMemberCluster: memberAwait,
+		username:              "community-user",
+		identityID:            uuid.New(),
+	}
+	createAppStudioUser(t, awaitilities, communityUser)
+	communityUserProxyClient, err := hostAwait.CreateAPIProxyClient(t, communityUser.token, hostAwait.APIProxyURL)
+	require.NoError(t, err)
+
+	t.Run("space is flagged as community", func(t *testing.T) {
+		// when
+		sb := CreateCommunitySpaceBinding(t, hostAwait, space.Name, space.Namespace)
+		require.NotNil(t, sb)
+
+		// Wait until space is flagged as community
+		require.NoError(t,
+			waitpoll.Poll(hostAwait.RetryInterval, hostAwait.Timeout, func() (bool, error) {
+				opts := []client.ListOption{
+					client.MatchingLabels{
+						toolchainv1alpha1.SpaceBindingMasterUserRecordLabelKey: toolchainv1alpha1.KubesawAuthenticatedUsername,
+						toolchainv1alpha1.SpaceBindingSpaceLabelKey:            space.Name,
+					},
+					client.InNamespace(space.Namespace),
+				}
+				sbs := &toolchainv1alpha1.SpaceBindingList{}
+				if err := hostAwait.Client.List(context.TODO(), sbs, opts...); err != nil {
+					return false, err
+				}
+
+				return len(sbs.Items) == 1, nil
+			}))
+
+		sp := toolchainv1alpha1.Space{}
+		err = hostAwait.Client.Get(context.TODO(), client.ObjectKeyFromObject(space), &sp)
+		require.NoError(t, err)
+
+		/*
+		   Given Space exists for user A
+		   Given User community-user exists
+		   When  A flags their space visibility to "community"
+		   Then  community-user can view A's Space
+		   And   community-user cannot create resources in A's Space
+		*/
+		t.Run("community user access to community space", func(t *testing.T) {
+			require.NotEmpty(t, sp.Status.ProvisionedNamespaces)
+
+			t.Run("community user can list config maps from community space", func(t *testing.T) {
+				// then
+				cms := corev1.ConfigMapList{}
+
+				communityUserProxyClient, err := hostAwait.CreateAPIProxyClient(t, communityUser.token, hostAwait.ProxyURLWithWorkspaceContext(sp.Name))
+				require.NoError(t, err)
+
+				err = communityUserProxyClient.List(context.TODO(), &cms, client.InNamespace(sp.Status.ProvisionedNamespaces[0].Name))
+				require.NoError(t, err)
+			})
+
+			t.Run("community user cannot create config maps into space", func(t *testing.T) {
+				cm := corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-cm",
+						Namespace: sp.Status.ProvisionedNamespaces[0].Name,
+					},
+				}
+				err := communityUserProxyClient.Create(context.TODO(), &cm)
+				require.Error(t, err)
```

Comment:
minor - but maybe we could check that the returned error is the expected one using `require.EqualError`

Reply:
done in 8d2872804d812e52a62d4144e4451b21db20e798